src/layer/mod.rs

```diff
@@ -10,23 +10,17 @@ use tracing_core::callsite;
 use tracing_subscriber::registry::LookupSpan;
 
 use crate::{
-    native::{EventWriter, ProviderTypes},
+    native::{OutputMode, ProviderTraits},
     statics::get_event_metadata,
 };
 
-pub(crate) struct _EtwLayer<S, Mode: ProviderTypes>
-where
-    Mode::Provider: crate::native::EventWriter<Mode> + 'static,
```

Comment:
These constraints were necessary only because the compiler can't look through nested classes to determine if constraints on the bottommost class are satisfied by the generics on the topmost instantiation. Each class must replicate the constraints all the way up the chain. Removing all of this makes the code easier to follow.